sql/core/src/main/scala/org/apache/spark/sql/execution/ShuffledRowRDD.scala

```diff
@@ -116,7 +116,8 @@ class CoalescedPartitioner(val parent: Partitioner, val partitionStartIndices: A
 class ShuffledRowRDD(
     var dependency: ShuffleDependency[Int, InternalRow, InternalRow],
     metrics: Map[String, SQLMetric],
-    specifiedPartitionStartIndices: Option[Array[Int]] = None)
+    specifiedPartitionStartIndices: Option[Array[Int]] = None,
```

Comment:
how about `Option[Array[(Int, Int)]]`?

Reply:
maybe the separate definition to `specifiedPartitionStartIndices` and `specifiedPartitionEndIndices` more clear？

Reply:
`Option[Array[(Int, Int)]]` is more type-safe. It eliminates problems like
1. `specifiedPartitionStartIndices` is specified but `specifiedPartitionEndIndices` is not.
2. these 2 have different lengths.